beacon/client/http-provider.go

```diff
@@ -36,24 +39,35 @@ const (
 	RequestWithdrawalCredentialsChangePath = "/eth/v1/beacon/pool/bls_to_execution_changes"
 
 	MaxRequestValidatorsCount = 600
+
+	fastGetMethod string = "Fast GET"
+	slowGetMethod string = "Slow GET"
+	postMethod    string = "POST"
```

Comment:
instead of `postMethod` there's `http.MethodPost`.

Instead of multiple clients, use https://pkg.go.dev/net/http#NewRequestWithContext which gives you control over the timeout via a [context with a timeout](https://pkg.go.dev/context#WithTimeout)

Pass the resulting `Request` to `client.Do` and pick a _large_ timeout for client.Timeout (at least the max of the timeouts specified in context.WithTimeout)

Better yet, use the provided `context.Context` in each of the getters unless nil, and default to an appropriate-lengthed timeout. It would be an inflexible abstraction if it prescribed a two-tier timeout system instead of just using `context.Context`